### PR TITLE
[Serve] Use serve_instance for test_http_headers to reduce the test time

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?
rewrite test to use shared instance instead of ray shutdown.
In my local, time is reduced from 40s to 23s.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by sihanwang41** [comment]
oh, acutally, it was slow because I use `ray_shutdown`, trying to use shared instance.

File: python/ray/serve/tests/test_http_headers.py [hunk 1]
```diff
@@ -24,50 +24,52 @@ def __call__(self):
...
 
 
-@pytest.mark.parametrize("deploy_type", ["basic", "fastapi", "starlette_resp"])
-def test_user_provided_request_id_header(ray_shutdown, deploy_type):
-    """Test that a user-provided request_id is propagated to the
-    replica and returned as a header."""
-
-    if deploy_type == "fastapi":
-        app = FastAPI()
+class TestUserProvidedRequestIDHeader:
```
Comment (sihanwang41) [inline python/ray/serve/tests/test_http_headers.py:27]: No change in test logic, just split up the test within a class and use the serve instance

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
please update PR title to be more descriptive

**Comment by sihanwang41** [comment]
@edoakes ping for merge. Windows tests pass.

**Pull request merged**